programs/zk-token-proof/src/lib.rs

```diff
@@ -40,24 +42,59 @@ where
     let transaction_context = &invoke_context.transaction_context;
     let instruction_context = transaction_context.get_current_instruction_context()?;
     let instruction_data = instruction_context.get_instruction_data();
-    let proof_data = ProofInstruction::proof_data::<T, U>(instruction_data).ok_or_else(|| {
-        ic_msg!(invoke_context, "invalid proof data");
-        InstructionError::InvalidInstructionData
-    })?;
+
+    // number of accessed accounts so far
+    let mut accessed_accounts = 0_u16;
+
+    // if instruction data is exactly 4 bytes, then read proof from an account
+    let proof_data = if instruction_data.len() == PROOF_OFFSET_LENGTH {
+        let proof_data_account = instruction_context
+            .try_borrow_instruction_account(transaction_context, accessed_accounts)?;
+        accessed_accounts = accessed_accounts.checked_add(1).unwrap();
+
+        let proof_data_offset = u32::from_le_bytes(
+            instruction_data[1..PROOF_OFFSET_LENGTH]
```

Comment:
What's the first byte for exactly?

Reply:
The first byte is the instruction enum discriminator. I added comments to clarify that.

Reply:
Oh darn, sorry, I thought I deleted this comment after reading through the code. Thanks for the explanation either way!